File: Containers/2. Matrices/3. Multiplication/1. Schoolbook Algorithm/Matrix Multiplication.cpp
#include<iostream>
#include<conio.h>

int main()
{
    using namespace std;

	int a[3][3],m,n, b[3][3],p,q, c[3][3], i,j,k;

	cout<<"\n Matrix A : \n";
	cout<<"\n Rows     : ";
	cin>>m;
	cout<<"\n Columns  : ";
	cin>>n;

	cout<<"\n Matrix B : \n";
	cout<<"\n Rows     : ";
	cin>>p;
	cout<<"\n Columns  : ";
	cin>>q;

	if(n==p)
	{
		cout<<"\n Matrix multiplication is possible \n";

		cout<<"\n Enter the elements of matrix A : \n\n";
		for(i=0;i<m;i++)
        {
            cout<<" ";
            for(j=0;j<n;j++)
            {
                cin>>a[i][j];
                for(k=0;k<=j;k++)
                    cout<<"\t";
            }
            cout<<"\n";
        }

        cout<<"\n Enter the elements of matrix B : \n\n";
        for(i=0;i<p;i++)
        {
            cout<<" ";
            for(j=0;j<q;j++)
            {
                cin>>b[i][j];
                for(k=0;k<=j;k++)
                    cout<<"\t";
            }
            cout<<"\n";
        }

        for(i=0;i<m;i++)
            for(j=0;j<q;j++)
            {
                c[i][j]=0;
                for(k=0;k<n;k++)
                    c[i][j]+=a[i][k]*b[k][j];
            }

        cout<<"\n Matrix A : \n\n";
        for(i=0;i<m;i++)
        {
            for(j=0;j<n;j++)
                cout<<" "<<a[i][j]<<"\t";
            cout<<"\n\n";
        }

        cout<<"\n Matrix B : \n\n";
        for(i=0;i<p;i++)
        {
            for(j=0;j<q;j++)
                cout<<" "<<b[i][j]<<"\t";
            cout<<"\n\n";
        }

        cout<<"\n Matrix C : \n\n";
        for(i=0;i<m;i++)
        {
            for(j=0;j<q;j++)
                cout<<" "<<c[i][j]<<"\t";
            cout<<"\n\n";
        }

	}

    else cout<<"\n Matrix multiplication not possible \n";
}

File: Containers/2. Matrices/3. Multiplication/1. Schoolbook Algorithm/Matrix Multiplication.py
import msvcrt
import os

a=[]
print "\n Matrix A : "
m=int(input("\n Rows     : "))
n=int(input("\n Columns  : "))

b=[]
print "\n Matrix B : "
p=int(input("\n Rows     : "))
q=int(input("\n Columns  : "))

if n==p:
    print("\n Matrix multiplication is possible")
    char=msvcrt.getch()
    os.system('cls')
else:
    print "\n Matrix multiplication not possible"
    char=msvcrt.getch()
    exit()
    
print("\n Enter the elements of matrix A : ")
for i in range(0,m):
    a.append([])
    for j in range(0,n):
        a[i].append(input())
    print(" ")

print("\n Enter the elements of matrix B : ")
for i in range(0,p):
    b.append([])
    for j in range(0,q):
        b[i].append(input())
    print(" ")

c=[]
for i in range(0,m):
    c.append([])
    for j in range(0,q):
        temp=0
        for k in range(0,n):
            temp+=a[i][k]*b[k][j]
        c[i].append(temp)

os.system('cls')

print("\n Matrix A : \n")
for i in range(0,m):
    print a[i],"\n"

print("\n Matrix B : \n")
for i in range(0,p):
    print b[i],"\n"

print("\n Matrix C : \n")
for i in range(0,m):
    print c[i],"\n"

char=msvcrt.getch()
